source/base/parameter_handler.cc

```diff
@@ -1176,6 +1203,9 @@ ParameterHandler::print_parameters(std::ostream &    out,
   // first
   if (style == XML)
     {
+      if (print_documentation == false)
+        simplify_xml_recursively(*current_entries);
```

Comment:
Don't you need to call this on `sorted_entries`? I may be misreading the code here, but it seems to me like you're permanently changing the state of the `entries` object...